### PR TITLE
Updated README with guided demos and videos

### PR DESCRIPTION
**Pull request by Maxusmusti**
# Issue link
#286 

# What changes have been made
Added guided demos and video to README

# Verification steps
Make sure the links all work, and that there are no typos

## Checks
- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [ ] Manual tests
   - [x] Testing is not required for this change

### CODE REVIEW

**Review by KPostOffice — APPROVED** [review]
/lgtm

File: README.md [hunk 1]
```diff
@@ -62,7 +65,7 @@ The following instructions apply when doing release manually. This may be requir
 
 * Check and update the version in "pyproject.toml" file.
 * Generate new documentation.
-`pdoc --html -o docs src/codeflare_sdk && pushd docs && rm -rf cluster job utils && mv codeflare_sdk/* . && rm -rf codeflare_sdk && popd` (it is possible to install **pdoc** using the following command `poetry install --with docs`)
+`pdoc --html -o docs src/codeflare_sdk && pushd docs && rm -rf cluster job utils && mv codeflare_sdk/* . && rm -rf codeflare_sdk && popd && find docs -type f -name "*.html" -exec bash -c "echo '' >> {}" \;` (it is possible to install **pdoc** using the following command `poetry install --with docs`)
```
Comment (anishasthana) [inline README.md:68]: Do we need to update the github release action too?

**Review by anishasthana — APPROVED** [review]
/lgtm

**Comment by anishasthana** [comment]
@MichaelClifford is the only person who can approve this

**Review by MichaelClifford — APPROVED** [review]

**Pull request merged**